.github/workflows/repo2cpe-download.yaml

```diff
@@ -0,0 +1,38 @@
+name: Download and Upload Repository Mappings
+
+on:
+  schedule:
+  - cron: "0 0 * * *"
+
+jobs:
+  download-and-upload:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Authenticate with Google Cloud
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: ${{ secrets.GOOGLE_SA_CIRCLECI_SCANNER }}
+
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v1
+
+    - name: Download repository mappings
+      run: |
+        mkdir redhat-repository-mappings
+        curl --fail --silent --show-error --max-time 60 --retry 3 --create-dirs \
+          -o 'redhat-repository-mappings/#1' \
+          'https://access.redhat.com/security/data/metrics/{repository-to-cpe.json,container-name-repos-map.json}'
+        for f in redhat-repository-mappings/*; do
+          jq empty "$f"
+        done
+
+    - name: Upload repository mappings to Google Cloud Storage
+      run: |
+        if ! gsutil cp -r "redhat-repository-mappings" "gs://scanner-v4-test/"; then
+          curl -X POST -H 'Content-type: application/json' --data '{"text":"<!subteam^S04S96AAVND|acs-scanner-primary> Failed to upload Redhat repository mapping to storage"}' ${{ secrets.SLACK_ONCALL_SCANNER_WEBHOOK }}
+          exit 1
+        fi
```

Comment:
Can we use [failure expressions](https://docs.github.com/en/actions/learn-github-actions/expressions#failure) for this?

Reply:
Errors during download of the files won't be captured by your proposed patch.

Reply:
if `gsutil cp -r "redhat-repository-mappings" "gs://scanner-v4-test/"` fails, there's a gsutil related error based on my test experiences 🤔️

And failure expressions didn't work with `secrets.SLACK_ONCALL_SCANNER_WEBHOOK` :  TypeError [ERR_INVALID_CHAR]: Invalid character in header content 
```
- name: Notify Slack
      env:
        SLACK_BOT_TOKEN: ${{ secrets.SLACK_ONCALL_SCANNER_WEBHOOK }}
      uses: slackapi/slack-github-action@v1.23.0
      with:
        slack-message: |
          The step to upload repository mappings to Google Cloud Storage has failed.
          Please @acs-scanner-primary for assistance.
channel-id: "C04S0D3EHKP"
```
["Authorization"]

Reply:
I am pointing out that the only time we will be notified is if the upload to gsutils fails. That means, we won't be notified if the download fails. Can you confirm?

Reply:
> And failure expressions didn't work with secrets.SLACK_ONCALL_SCANNER_WEBHOOK : TypeError [ERR_INVALID_CHAR]: Invalid character in header content ["Authorization"]

I don't see the conditional `if:` in your example above. Maybe what Gavin suggested was using `curl` inside a  `run:` script, but it should be conditioned to the job failing, hence the need for a "fail expression" I mentioned in the docs.

Reply:
> 

This was the whole block that I was testing:
```
- name: Upload repository mappings to Google Cloud Storage
      run: |
        gsutil cp -r "whats-so-funny" "gs://scanner-v4-test/"
      if: failure()

    - name: Notify Slack
      uses: slackapi/slack-github-action@v1.23.0
      with:
        channel-id: C04S0D3EHKP
        slack-token: ${{ secrets.SLACK_ONCALL_SCANNER_WEBHOOK }}
        slack-message: |
          The step to upload repository mappings to Google Cloud Storage has failed.
          Please contact @acs-scanner-primary for assistance.
```
And `secrets.SLACK_ONCALL_SCANNER_WEBHOOK` wasn't working this way and we had a discussion about this on Slack: https://redhat-internal.slack.com/archives/CC5UHD0KA/p1688058097822359?thread_ts=1688054155.470279&cid=CC5UHD0KA